conan/tools/microsoft/visual.py

```diff
@@ -54,11 +54,14 @@ def generate(self, scope="build"):
 
 
 def vs_ide_version(conanfile):
+    """
+    Get the VS IDE version as string
+    """
     compiler = conanfile.settings.get_safe("compiler")
     compiler_version = (conanfile.settings.get_safe("compiler.base.version") or
                         conanfile.settings.get_safe("compiler.version"))
     if compiler == "msvc":
-        toolset_override = conanfile.conf["tools.microsoft.msbuild:vs_version"]
+        toolset_override = conanfile.conf.get("tools.microsoft.msbuild:vs_version", check_type=str)
```

Comment:
@memsharded I have added a kind of several smart conversions for `str` and `bool` types so, if you specify the `check_type`, they'll help the user to ensure that the value gotten is a string or a boolean one.